sql/dd/impl/sdi.cc

```diff
@@ -387,57 +388,33 @@ bool CheckDefaultCompatibility(const RJ_Document &doc) {
              mysqld_version_id.GetUint64(), std::uint64_t(MYSQL_VERSION_ID));
     return true;
   }
+#endif /* !XTRABACKUP */
 
-<<<<<<< HEAD
-  if (doc.HasMember("mysqld_version_id")) {
-#if !defined(XTRABACKUP)
-    RJ_Value &mysqld_version_id = doc["mysqld_version_id"];
-    DBUG_ASSERT(mysqld_version_id.IsUint64());
-    if (mysqld_version_id.GetUint64() > std::uint64_t(MYSQL_VERSION_ID)) {
-      // Cannot deserialize SDIs from newer versions. Required?
-      my_error(ER_IMP_INCOMPATIBLE_MYSQLD_VERSION, MYF(0),
-               mysqld_version_id.GetUint64(), std::uint64_t(MYSQL_VERSION_ID));
-      return true;
-    }
-#endif
-  } else {
-    DBUG_ASSERT(false);
-  }
-
-  DBUG_ASSERT(doc.HasMember("dd_version"));
-  const RJ_Value &dd_version_val = doc["dd_version"];
-  DBUG_ASSERT(dd_version_val.IsUint());
-#if !defined(XTRABACKUP)
-=======
   assert(doc.HasMember("dd_version"));
   const RJ_Value &dd_version_val = doc["dd_version"];
   assert(dd_version_val.IsUint());
->>>>>>> mysql-8.0.24
+#if !defined(XTRABACKUP)
   uint dd_version = dd_version_val.GetUint();
   if (dd_version != Dictionary_impl::get_target_dd_version()) {
     // Incompatible change
     my_error(ER_IMP_INCOMPATIBLE_DD_VERSION, MYF(0), dd_version,
              Dictionary_impl::get_target_dd_version());
     return true;
   }
-#endif
+#endif /* !XTRABACKUP */
```

Comment:
are we suppressing the DD version check during SDI deserialization? (ie during prepare operation)?

Reply:
as per slack conversation. PXB skip checking if mysql_version / dd/ sdi of source are higher than pxb version. We let the code assert on later stage.